hayanime

```diff
@@ -320,29 +321,30 @@ hist_handler(){
 		query_result="${hist_data%%¢*}#"
 		ani_searched="${hist_data##"${hist_data%%¢*}"¢}" ani_searched="${ani_searched%%€*}"
 		ani_selected="1" qual_stream="high" query_count="1"
-		epi_selected="${hist_data#*€}" epi_selected="${epi_selected%%£*}" epi_selected="$(( ${epi_selected%%/*} + ${solv:-0} ))"
+		epi_selected="${hist_data#*€}" epi_selected="${epi_selected%%£*}"
+		printf '%s' "${epi_selected%%/*}" | grep -q '[0-9]*\.[0-9]*' && epi_selected="${epi_selected%%/*}" || epi_selected="$(( ${epi_selected%%/*} + ${solv:-0} ))"
 		player="${hist_data##*£}"
 	fi
 }
 
 hist_update(){
-	hist_count="$(awk 'BEGIN{c=0} //{c++} END{print c}' ~/"${hist_file}")"
+	[ -e ~/"${hist_file}".tmp ] && rm -f ~/"${hist_file}".tmp
+	hist_count="$(grep -c '[^[:space:]]' ~/"${hist_file}")"
 	[ "${hist_count}" = "0" ] && die 1 "Theres no to update, History is Empty"
 	printf "\033[38;5;3m\033[2K\r%b\033[0m" "Checking..."
 	curl -sLfk "${base_url}/v1/spy-x-family" -A "${user_agent}" > /dev/null 2>&1 || die 1 "\033[2K\rServer can't be reach..."
-	cp ~/"${hist_file}" ~/"${hist_file}".tmp
 	while read -r entry; do
 		: "$((entriter += 1))"
 		cur_ep="${entry#*€}" cur_ep="${cur_ep%%£*}" cur_ep="${cur_ep%%/*}"
-		(
-			tot_ep="$(curl -sLk "${base_url}${entry%%¢*}" -A "${user_agent}" | sed -nE 's_.*eptotal":([^\,\}]*)(,.*|\}</.*)_\1_p')"
-			printf '%s\n' "$(sed "${entriter}s%\(.*€\)\(.*\)\(£.*\)%\\1${cur_ep}/${tot_ep}\\3%" ~/"${hist_file}".tmp)" > ~/"${hist_file}".tmp
-		) &
-		[ "$(jobs -p | awk 'BEGIN{c=0} //{c++} END{print c}')" -gt 2 ] && wait "${!}"
+		tot_ep="$(curl -sLk "${base_url}${entry%%¢*}" -A "${user_agent}" | sed -nE 's_.*eptotal":([^\,\}]*)(,.*|\}</.*)_\1_p')"
+		printf '%s\n' "${entry}" | sed "s%\(.*€\)\(.*\)\(£.*\)%\\1${cur_ep}/${tot_ep}\\3%" >> ~/"${hist_file}".tmp
```

Comment:
We're going back to iterate one-by-one history updating [again], there was a chance that it deletes all the contents of history while updating, I don't want to deal with it lol.